build_tools/update_translations.fish

```diff
@@ -71,10 +78,23 @@ if set -l --query extract
     or exit 1
 end
 
+if set -l --query _flag_dry_run
+    # On a dry run, we do not modify po/ but write to a temporary directory instead and check if
+    # there is a difference between po/ and the tmpdir after re-generating the PO files.
+    set -g tmpdir (mktemp -d)
+    # On a dry-run, we do not update the MO files.
+    set -l --erase mo
+end
+
 for po_file in $po_files
+    if set -g --query tmpdir
```

Comment:
we should probably protect against a user exporting `tmpdir`
(`tmpdir=bad build_tools/update_translations.fish`)
imported variables are always global, so better reset it here.

Of course this is probably a user error that might cause problems
elsewhere too but should prefer to be predictable